src/System.Diagnostics.StackTrace/src/System/Diagnostics/StackTraceSymbols.CoreCLR.cs

```diff
@@ -0,0 +1,244 @@
+                        }
+                    }
+                }
+            }
+        }
+
+        /// <summary>
+        /// Returns the portable PDB reader for the assembly path
+        /// </summary>
+        /// <param name="assemblyPath">file path of the assembly or null</param>
+        /// <param name="loadedPeAddress">loaded PE image address or zero</param>
+        /// <param name="loadedPeSize">loaded PE image size</param>
+        /// <param name="inMemoryPdbAddress">in memory PDB address or zero</param>
+        /// <param name="inMemoryPdbSize">in memory PDB size</param>
+        /// <param name="reader">returns the reader</param>
+        /// <returns>reader</returns>
+        private MetadataReader GetReader(string assemblyPath, IntPtr loadedPeAddress, int loadedPeSize, IntPtr inMemoryPdbAddress, int inMemoryPdbSize)
+        {
+            if (loadedPeAddress != IntPtr.Zero)
+            {
+                Tuple<MetadataReaderProvider, MetadataReader> tuple;
+                if (_readerCache.TryGetValue(loadedPeAddress, out tuple))
+                {
+                    return tuple.Item2;
+                }
+            }
+
+            MetadataReaderProvider provider = null;
+            MetadataReader reader = null;
+
+            if (assemblyPath != null)
+            {
+                uint stamp;
+                int age;
+                Guid guid;
+
+                string pdbName = GetPdbPathFromPeStream(assemblyPath, loadedPeAddress, loadedPeSize, out age, out guid, out stamp);
+                if (pdbName != null)
+                {
+                    if (File.Exists(pdbName))
+                    {
+                        Stream pdbStream = File.OpenRead(pdbName);
+
+                        provider = MetadataReaderProvider.FromPortablePdbStream(pdbStream);
+                        MetadataReader rdr = provider.GetMetadataReader();
+
+                        // Validate that the PDB matches the assembly version
+                        if (age == 1 && IdEquals(rdr.DebugMetadataHeader.Id, guid, stamp))
+                        {
+                            reader = rdr;
+                        }
+                        else
+                        {
+                            provider.Dispose();
+                            provider = null;
+                        }
+                    }
+                }
+            }
+            else if (inMemoryPdbAddress != IntPtr.Zero && inMemoryPdbSize > 0)
+            {
+                unsafe
+                {
+                    provider = MetadataReaderProvider.FromPortablePdbImage((byte*)inMemoryPdbAddress.ToPointer(), inMemoryPdbSize);
+                    reader = provider.GetMetadataReader();
+                }
+            }
+
+            if (reader != null && loadedPeAddress != IntPtr.Zero)
+            {
+                _readerCache.Add(loadedPeAddress, new Tuple<MetadataReaderProvider, MetadataReader>(provider, reader));
```

Comment:
Nit: what you have is fine, but it could also be shortened with ```Tuple.Create``` in order to get type inference and not need to specify the generic type args